internal/backend.go

```diff
@@ -72,6 +76,36 @@ func Run(verbose bool) {
 	select {}
 }
 
+func setupExitHandler() {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		sig := <-sigs
+		log.Printf("Received signal %d, trying to restore fan settings...", sig)
+
+		for _, controller := range Controllers {
+			for _, fan := range controller.Fans {
+				if fan.Config != nil {
+					// try to reset the pwm_enabled value
+					if fan.OriginalPwmEnabled != 1 {
+						err := setPwmEnabled(fan, fan.OriginalPwmEnabled)
+						if err == nil {
+							continue
+						}
+					}
+					err := setPwm(fan, MaxPwmValue)
+					if err != nil {
+						log.Printf("WARNING: Unable to revert fan %s, make sure it is running!", fan.Config.Id)
+					}
+				}
+			}
+		}
+		log.Println("Exiting...")
+		os.Exit(0)
```

Comment:
This is simply doing explicitly what Go already does implicitly. `os.Exit(0)` is the default.

Reply:
Right, but when is it doing that? I put it there to force it after the signal handler is done. Will go call `os.Exit` right after the signal handler returns?